Common/Chairloader/ConfigNode.h

```diff
@@ -0,0 +1,247 @@
+//
+// Created by theli on 4/25/2025.
+//
+
+#ifndef CHAIRLOADER_CONFIGNODE_H
+#define CHAIRLOADER_CONFIGNODE_H
+
+#include <optional>
+#include <stdexcept>
+#include <variant>
+
+enum class NodeType { String, Bool, Int, Uint, Int64, Uint64, Float, XmlNode, Enum, Unknown };
+
+inline NodeType toNodeType(std::string_view s) {
+    if      (s=="string")   return NodeType::String;
+    else if (s=="bool")     return NodeType::Bool;
+    else if (s=="int")      return NodeType::Int;
+    else if (s=="uint")     return NodeType::Uint;
+    else if (s=="int64")    return NodeType::Int64;
+    else if (s=="uint64")   return NodeType::Uint64;
+    else if (s=="float")    return NodeType::Float;
+    else if (s=="xmlnode")  return NodeType::XmlNode;
+    else if (s=="enum")     return NodeType::Enum;
+    else                    return NodeType::Unknown;
+}
+
+inline std::string toString(NodeType type) {
+    switch(type) {
+        case NodeType::String:   return "string";
+        case NodeType::Bool:     return "bool";
+        case NodeType::Int:      return "int";
+        case NodeType::Uint:     return "uint";
+        case NodeType::Int64:    return "int64";
+        case NodeType::Uint64:   return "uint64";
+        case NodeType::Float:    return "float";
+        case NodeType::XmlNode:  return "xmlnode";
+        case NodeType::Enum:     return "enum";
+        default:                 return "unknown";
+    }
+}
+
+
+class ConfigNode {
```

Comment:
A small comment describing what a config node is would be nice. I assume this is a single config option that may contain:
- single int/float/bool/string, or
- raw XML node with mod-specific format, or
- multiple other nodes (for nested options/config sections)

It also looks like `ConfigNode` is a wrapper around `pugi::xml_node`, but it also caches info like name and description